Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -527,6 +530,106 @@ class Gx2Fitter {
 
           // We count the processed state
           ++result.processedStates;
+
+          // Update the number of holes count only when encountering a
+          // measurement
+          result.measurementHoles = result.missedActiveSurfaces.size();
+        } else if (surface->associatedDetectorElement() != nullptr ||
+                   surface->surfaceMaterial() != nullptr) {
+          // Here we handle material and holes
+          // TODO add material handling
+          ACTS_VERBOSE("Non-Measurement surface " << surface->geometryId()
+                                                  << " detected.");
+
+          // We only create track states here if there is already a measurement
+          // detected or if the surface has material (no holes before the first
+          // measurement)
+          if (result.measurementStates > 0
+              // || surface->surfaceMaterial() != nullptr
+          ) {
+            ACTS_VERBOSE("Handle hole.");
+
+            auto& fittedStates = *result.fittedStates;
+
+            // Mask for the track states. We don't need Smoothed and Filtered
+            TrackStatePropMask mask =
+                ~(TrackStatePropMask::Smoothed | TrackStatePropMask::Filtered);
+
+            ACTS_VERBOSE("    processSurface: addTrackState");
+
+            // Add a <mask> TrackState entry multi trajectory. This allocates
+            // storage for all components, which we will set later.
+            typename traj_t::TrackStateProxy trackStateProxy =
+                fittedStates.makeTrackState(mask, result.lastTrackIndex);
+            std::size_t currentTrackIndex = trackStateProxy.index();
+            {
+              // Set the trackStateProxy components with the state from the
+              // ongoing propagation
+              {
+                trackStateProxy.setReferenceSurface(surface->getSharedPtr());
+                // Bind the transported state to the current surface
+                auto res = stepper.boundState(state.stepping, *surface, false,
+                                              freeToBoundCorrection);
```

Comment:
`false` means we do not transport the covariance - is this intended?

Reply:
yes, I checked again, this is intended.

Reply:
hm but at the same time you use the jacobian and covariance further down. they will not be valid if you do not transport them

Reply:
As we discovered, we won't need the explicit covariance transport, since we already have it transported in:

https://github.com/acts-project/acts/blob/23d6a0373086082c2c4414ec288d94f36f717615/Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp#L431